seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/sink/bucket/PaimonBucketAssigner.java

```diff
@@ -59,27 +65,85 @@ public PaimonBucketAssigner(Table table, int numAssigners, int assignId) {
 
     private void loadBucketIndex(FileStoreTable fileStoreTable, int numAssigners, int assignId) {
         IndexBootstrap indexBootstrap = new IndexBootstrap(fileStoreTable);
+        List<String> fieldNames = schema.fieldNames();
+        Map<String, Integer> fieldIndexMap =
+                IntStream.range(0, fieldNames.size())
+                        .boxed()
+                        .collect(Collectors.toMap(fieldNames::get, Function.identity()));
+        List<DataField> primaryKeys = schema.primaryKeysFields();
         try (RecordReader<InternalRow> recordReader =
                 indexBootstrap.bootstrap(numAssigners, assignId)) {
             RecordReaderIterator<InternalRow> readerIterator =
                     new RecordReaderIterator<>(recordReader);
             while (readerIterator.hasNext()) {
                 InternalRow row = readerIterator.next();
-                assign(row);
+                GenericRow binaryRow = new GenericRow(fieldNames.size());
+                for (int i = 0; i < primaryKeys.size(); i++) {
+                    String name = primaryKeys.get(i).name();
+                    DataType type = primaryKeys.get(i).type();
+                    binaryRow.setField(
+                            fieldIndexMap.get(name), getPrimaryFieldValue(row, i, type, name));
```

Comment:
You are no need define the method named getPrimaryFiledValue. You can use it like this InternalRow.createFieldGetter(type,i).getFieldOrNull(binaryRow);

Reply:
Otherwise the primary key  of paimon support all data types?

Reply:
get